include/sys/util.h

```diff
@@ -711,6 +711,21 @@ u8_t u8_to_dec(char *buf, u8_t buflen, u8_t value);
 
 #define UTIL_IF(c) UTIL_IIF(UTIL_BOOL(c))
 
+/*
+ * These are like || and &&, but they do evaluation and
+ * short-circuiting at preprocessor time instead of runtime.
+ *
+ * UTIL_OR(foo, bar) is sometimes a replacement for (foo || bar)
+ * when "bar" is an expression that would cause a build
+ * error when "foo" is true.
+ *
+ * UTIL_AND(foo, bar) is sometimes a replacement for (foo && bar)
+ * when "bar" is an expression that would cause a build
+ * error when "foo" is false.
+ */
+#define UTIL_OR(a, b) COND_CODE_1(UTIL_BOOL(a), (a), (b))
```

Comment:
shouldn't it be ` COND_CODE_1(UTIL_BOOL(a), UTIL_BOOL(a),UTIL_BOOL(b))`?

Reply:
I wanted them to return the actual nonzero value for things like this:

```
#define DT_NUM_INST(compat)					\
	UTIL_AND(DT_HAS_COMPAT(compat),				\
		 UTIL_CAT(DT_N_INST, DT_DASH(compat, NUM)))
```

Think it's worth changing?